tapdb/universe.go

```diff
@@ -612,13 +615,27 @@ func universeFetchProofLeaf(ctx context.Context,
 
 // MintingKeys returns all the keys inserted in the universe.
 func (b *BaseUniverseTree) MintingKeys(ctx context.Context,
+	direction universe.SortDirection, offset, limit int,
 ) ([]universe.LeafKey, error) {
 
 	var leafKeys []universe.LeafKey
 
 	readTx := NewBaseUniverseReadTx()
 	dbErr := b.db.ExecTx(ctx, &readTx, func(db BaseUniverseStore) error {
-		universeKeys, err := db.FetchUniverseKeys(ctx, b.smtNamespace)
+		universeKeys, err := db.FetchUniverseKeys(
+			ctx, UniverseLeafKeysQuery{
+				Namespace:     b.smtNamespace,
+				SortDirection: sqlInt16(direction),
+				NumOffset:     int32(offset),
+				NumLimit: func() int32 {
+					if limit == 0 {
```

Comment:
Discussed offline that this should default to the maximum page size, not `maxint32`. so an old client gets only the first page, not as much data as possible.